build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1288,6 +1315,218 @@ message WaitExecutionRequest {
   string name = 1;
 }
 
+// A request message for
+// [ActionCache.GetPotentialActionResults][build.bazel.remote.execution.v2.ActionCache.GetPotentialActionResults]
+// two-phase cache lookups.
+message GetPotentialActionResultsRequest {
+  // The instance of the execution system to operate against. A server may
+  // support multiple instances of the execution system (with their own workers,
+  // storage, caches, etc.). The server MAY require use of this field to select
+  // between them in an implementation-defined fashion, otherwise it can be
+  // omitted.
+  string instance_name = 1;
+
+  // The hash of an [Action][build.bazel.remote.execution.v2.Action] that acts as a weak fingerprint
+  // key to look up potential action results that must be compared to the client's current filesystem
+  // contents to get a match. On a match, the corresponding build outputs in the form of an
+  // [ActionResult][build.bazel.remote.execution.v2.ActionResult] can be retrieved from
+  // [ActionCache.GetActionResult][build.bazel.remote.execution.v2.ActionCache.GetActionResult].
+  //
+  // In two-phase caching an `Action` contains a normal reference to a
+  // [Command][build.bazel.remote.execution.v2.Command] but the `input_root_digest`
+  // can refer to a subset of the files that are predicted to be needed to run the `Action`.
+  // The input root can contain all files needed for the command, in which case the weak
+  // fingerprint should lead to just one output (or none if there are no cache hits).
+  // When the input root contains fewer files than those actually needed by the action,
+  // the weak fingerprint gets less specific, and the number of possible matches grows.
+  // It's possible for the input root to contain zero files, which is a very weak
+  // fingerprint as it can match any output that matches the `Command`.
+  // Less-specific weak fingerprints lead to more I/O and time spent matching pathsets against
+  // the client filesystem; adding in any predictions about the filesystem "focuses"
+  // the lookup to more-sepcific results.
```

Comment:
>// the lookup to more-sepcific results. [](start = 2, length = 39)

Does size of blob in Digest matter here? Should we use string only?

Reply:
Size needed: The weak FP implies creating a canonicalized blob with a size, compare that too

---
In reply to: [453807983](https://github.com/erikma/remote-apis/pull/1#discussion_r453807983) [](ancestors = 453807983)

Reply:
Plus note it's part of the URI for this API.

---
In reply to: [453808788](https://github.com/erikma/remote-apis/pull/1#discussion_r453808788) [](ancestors = 453808788,453807983)